### PR TITLE
[Improve](func)support array to window-func first-last-value arg type

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

support array to window-func first-last-value arg type 
eg. select first-value(array) over(xxx) 

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/utils/template_helpers.hpp [hunk 1]
```diff
@@ -67,7 +67,8 @@
     NUMERIC_TYPE_TO_COLUMN_TYPE(M)   \
     DECIMAL_TYPE_TO_COLUMN_TYPE(M)   \
     STRING_TYPE_TO_COLUMN_TYPE(M)    \
-    TIME_TYPE_TO_COLUMN_TYPE(M)
+    TIME_TYPE_TO_COLUMN_TYPE(M)      \
+    COMPLEX_TYPE_TO_COLUMN_TYPE(M)
 
 #define TYPE_TO_COLUMN_TYPE(M)   \
     TYPE_TO_BASIC_COLUMN_TYPE(M) \
```
Comment (zhangstar333) [inline be/src/vec/utils/template_helpers.hpp:74]: seems could use this macro dirctly in be/src/vec/aggregate_functions/aggregate_function_window.cpp

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java [hunk 2]
```diff
@@ -1348,6 +1348,38 @@ private void initAggregateBuiltins() {
...
                             t, "", "", "", "", "", true, false, true, true));
+
+            //first_value/last_value for array
+            addBuiltin(AggregateFunction.createAnalyticBuiltin("first_value",
+                    Lists.newArrayList(new ArrayType(t)), new ArrayType(t), Type.ARRAY,
+                    "",
+                    "",
+                    null,
+                    "",
+                    ""));
```
Comment (zhangstar333) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:1359]: this register could remove, now all function are vectorized, no need for row batch

File: be/src/vec/utils/template_helpers.hpp (hunk 1 above)
Comment (amorynan) [reply be/src/vec/utils/template_helpers.hpp:74]: ok I will just use TYPE_TO_COLUMN_TYPE(M)

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java (hunk 2 above)
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java:1359]: done!

File: be/src/vec/utils/template_helpers.hpp (hunk 1 above)
Comment (amorynan) [reply be/src/vec/utils/template_helpers.hpp:74]: done!

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 42.61 seconds
 stream load tsv:          453 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          69.1 seconds inserted 10000000 Rows, about 144K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230627070151_clickbench_pr_168533.html

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run clickbench

**Review by zhangstar333 — APPROVED** [review]

**Comment by amorynan** [comment]
run clickbench

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.72 seconds
 stream load tsv:          452 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          68.0 seconds inserted 10000000 Rows, about 147K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230627104738_clickbench_pr_168738.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.61 seconds
 stream load tsv:          451 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          73.4 seconds inserted 10000000 Rows, about 136K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230627114902_clickbench_pr_168794.html

**Comment by amorynan** [comment]
run p1

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**